pynetbox_data_uploader/lib/user_methods/csv_to_netbox.py

```diff
@@ -24,63 +27,96 @@ def __init__(self, url: str, token: str):
         :param token: The Netbox auth token.
         """
         self.netbox = NetboxConnect(url, token).api_object()
-        self.format_dict = FormatDict(self.netbox)
         self.exist = NetboxCheck(self.netbox)
         self.create = NetboxCreate(self.netbox)
+        self.query_dataclass = QueryDataclass(self.netbox)
 
-    def read_csv(self, file_path) -> List:
+    @staticmethod
+    def check_file_path(file_path: str):
+        """
+        This method checks if the filepath is valid. Raises an exception if it's invalid.
+        :param file_path: The path to the csv file.
+        """
+        print("Checking filepath...")
+        valid = Path(file_path).exists()
+        if not valid:
+            raise FileNotFoundError(f"Filepath: {file_path} is not valid.")
+        print("Filepath valid.")
+
+    @staticmethod
+    def read_csv(file_path: str) -> List[Device]:
         """
         This method calls the csv_to_python and seperate_data method.
         This will take the csv file and return a list of device dictionaries.
         :param file_path: The file path to the csv file to be read.
         :return: Returns a list of devices
         """
         print("Reading CSV...")
-        try:
-            pathlib.Path(file_path).exists()
-        except FileNotFoundError:
-            raise Exception("The given path is not valid.", FileNotFoundError)
-        device_data = self.format_dict.csv_to_python(file_path)
-        device_list = self.format_dict.separate_data(device_data)
+        dict_reader_class = open_file(file_path)
+        device_list = separate_data(dict_reader_class)
         print("Read CSV.")
         return device_list
 
-    def check_netbox(self, device_list: List) -> bool:
+    def check_netbox_device(self, device_list: List[Device]) -> bool:
         """
         This method calls the check_device_exists and check_device_type_exists method on each device in the list.
         :param device_list: A list of devices.
         :return: Returns True if the devices don't exist and device types do exist. Raises an Exception otherwise.
         """
         print("Checking devices in Netbox...")
         for device in device_list:
-            device_exist = self.exist.check_device_exists(device["name"])
+            device_exist = self.exist.check_device_exists(device.name)
             if device_exist:
-                raise Exception(f'Device {device["name"]} already exists in Netbox.')
-            type_exist = self.exist.check_device_type_exists(device["device_type"])
-            if not type_exist:
-                raise Exception(f'Type {device["device_type"]} does not exist.')
+                raise Exception(f"Device {device.name} already exists in Netbox.")

```

Comment:
please specify the type of exception you're raising - `Exception` is too generic - I recommend something like RuntimeError. 

Better yet - I would recommend writing your own error class - future PR

Reply:
👍